src/overlays/effects/ovl_Effect_Ss_Kakera/z_eff_ss_kakera.c

```diff
@@ -8,50 +8,355 @@
 
 #define PARAMS ((EffectSsKakeraInitParams*)initParamsx)
 
-s32 EffectSsKakera_Init(GlobalContext* globalCtx, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsKakera_Init(GlobalContext* globalCtx, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsKakera_Update(GlobalContext* globalCtx, u32 index, EffectSs* this);
 void EffectSsKakera_Draw(GlobalContext* globalCtx, u32 index, EffectSs* this);
 
-#if 0
+void func_8097E130(EffectSs* this, GlobalContext* globalCtx);
+f32 func_8097E368(f32, s32);
+f32 func_8097E384(f32, s32);
+f32 func_8097E3C0(f32, s32);
+f32 func_8097E400(f32, s32);
+
 const EffectSsInit Effect_Ss_Kakera_InitVars = {
     EFFECT_SS_KAKERA,
     EffectSsKakera_Init,
 };
+KakeraColorStruct D_8097EAD8[] = { { 0, { 255, 255, 255 } }, { 0, { 235, 170, 130 } }, { 1, { 210, 190, 170 } } };
+f32 D_8097EAE4[] = { 1.0f, 100.0f, 40.0f, 5.0f, 100.0f, 40.0f, 5.0f, 100.0f, 40.0f, 5.0f };
+f32 D_8097EB0C[] = { 0.05f, 1.0f };
+f32 D_8097EB14[] = { 4.0f, 0.1f, 0.3f, 0.9f, -0.1f, -0.3f, -0.9f };
+f32 D_8097EB30[] = { 0.1f, 1.0f, 6.0f };
+f32 (*D_8097EB3C[])(f32, s32) = { func_8097E368, func_8097E384, func_8097E384, func_8097E384, func_8097E3C0,
+                                  func_8097E3C0, func_8097E3C0, func_8097E400, func_8097E400, func_8097E400 };
+f32 D_8097EB64[] = { 10.0f, 20.0f, 40.0f };
+
+u32 EffectSsKakera_Init(GlobalContext* globalCtx, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsKakeraInitParams* params = PARAMS;
+
+    this->pos = params->pos;
+    this->velocity = params->velocity;
+    this->life = params->life;
+    this->priority = 101;
+    if (params->dList != NULL) {
+        this->gfx = params->dList;
+        switch (params->objId) {
+            case 1:
+            case 2:
+            case 3:
+                this->regs[10] = KAKERA_OBJECT_DEFAULT;
+                break;
+            default:
+                this->regs[10] = params->objId;
+                func_8097E130(this, globalCtx);
+                break;
+        }
+    } else {
+        __assert("../z_eff_kakera.c", 193);
+    }
+    this->draw = EffectSsKakera_Draw;
+    this->update = EffectSsKakera_Update;
+    this->vec = params->unk_18;
+    this->regs[0] = params->unk_2C;
+    this->regs[1] = params->gravity;
+    this->regs[2] = Rand_ZeroOne() * 32768.0f;
+    this->regs[3] = Rand_ZeroOne() * 32768.0f;
+    this->regs[4] = params->unk_26;
+    this->regs[5] = params->unk_28;
+    this->regs[6] = params->unk_2A;
+    this->regs[7] = params->scale;
+    this->regs[8] = params->unk_30;
+    this->regs[9] = params->unk_32;
+    this->regs[12] = params->colorIdx;
+    return 1;
+}
+
+f32 func_8097DE30(f32 center, f32 range) {
+    return (2.0f * (Rand_ZeroOne() * range) - range) + center;
+}
+
+void EffectSsKakera_Draw(GlobalContext* globalCtx, u32 index, EffectSs* this) {
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s16 pad;
+    f32 scale = this->regs[7] / 256.0f;
+    s32 color_index = this->regs[12];
```

Comment:
We don't use snake_case to name variables. I think the only exception are the autogenerated variables like the ones mips_to_c produces
```suggestion
    s32 colorIndex = this->regs[12];
```